lib/operations/create_collection.js

```diff
@@ -38,23 +51,21 @@ class CreateCollectionOperation extends OperationBase {
 
     let Collection = loadCollection();
 
-    // Get the write concern options
-    const finalOptions = applyWriteConcern(Object.assign({}, options), { db }, options);
-
     // Did the user destroy the topology
     if (db.serverConfig && db.serverConfig.isDestroyed()) {
       return callback(new MongoError('topology was destroyed'));
     }
 
-    const listCollectionOptions = Object.assign({}, finalOptions, { nameOnly: true });
+    let listCollectionOptions = Object.assign({}, options, { nameOnly: true });
```

Comment:
It looks like the order of constructing the command is reversed (ex: applyWriteConcern happening after merging some of the options). Is there a reason for this?

Reply:
This is because `applyWriteConcern` is only being performed on `listCollectionOptions`, since `CommandOperation` takes care of the write concern during `execute`.